### DISCUSSION TITLE
Directus 8 and MySQL 8

### DISCUSSION THREAD

**Discussion by Chris2Brooks** [directus/directus#2806] [Archive]
Hey all, just want to double check.

Does Directus 8.8.0 work well with MySQL 8?
Server: PHP 7.2.24

I'm planning on connecting Directus to a managed database by Digital Ocean but they only offer MySQL 8. 
Just want to hear your thoughts before I proceed to setting that up.

**Comment by davideperozzi** [comment]
@Chris2Brooks Nope unfortunately it doesn't work (at least for me). I'm using DigitalOcean too and get this error after migrating to 8.8.1 (docker):
![image](https://user-images.githubusercontent.com/4140184/90070587-69993280-dcf4-11ea-829f-8e8618f5ee4b.png)

**Comment by sneakersgames** [comment]
I confirm having the same issue and I would stay away from Directus for production environments running MySQL8

**Comment by sneakersgames** [comment]
@davideperozzi try changing the default SQL modes and removing `ANSI_QUOTES`

**Comment by rijkvanzanten** [comment]
> Does Directus 8.8.0 work well with MySQL 8?

No, Directus 8.8.0 hasn't been built to work with MySQL 8 specifically. While it _might_ run when changing certain settings on the DB side (see above), there's no official support.

(v9 will support MySQL 8 (as well as 5.7, postgres, sqlite, ms sql, and oracledb) out of the box)

**Reply by botzill** [reply]
Hi @rijkvanzanten. Do you have any updated on v9? I see there is alpha3 release there, can we test it?

**Reply by rijkvanzanten** [reply]
Yeah, feel free to take it for a spin: https://github.com/directus/next

**Reply by gusdecool** [reply]
@rijkvanzanten the link to https://github.com/directus/next are 404

**Reply by joselcvarela** [reply]
Hello @gusdecool 
This repo https://github.com/directus/directus is now for the v9 version.
For the v8 version you can check here https://github.com/directus/v8-archive

**Comment by botzill** [comment]
is there a docker image for this version?

**Reply by gusdecool** [reply]
I don't think it's available yet. I tried the `9.0.0-rc.95` which was built 9 days ago and still got error

```
directus  | 14:41:34 🚨 Can't connect to the database.
directus  | 14:41:34 🚨 ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client 
directus  | Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client
directus  |     at Handshake.Sequence._packetToError (/directus/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
directus  |     at Handshake.ErrorPacket (/directus/node_modules/mysql/lib/protocol/sequences/Handshake.js:123:18)
directus  |     at Protocol._parsePacket (/directus/node_modules/mysql/lib/protocol/Protocol.js:291:23)
directus  |     at Parser._parsePacket (/directus/node_modules/mysql/lib/protocol/Parser.js:433:10)
directus  |     at Parser.write (/directus/node_modules/mysql/lib/protocol/Parser.js:43:10)
directus  |     at Protocol.write (/directus/node_modules/mysql/lib/protocol/Protocol.js:38:16)
directus  |     at Socket.<anonymous> (/directus/node_modules/mysql/lib/Connection.js:88:28)
directus  |     at Socket.<anonymous> (/directus/node_modules/mysql/lib/Connection.js:526:10)
directus  |     at Socket.emit (node:events:390:28)
directus  |     at addChunk (node:internal/streams/readable:315:12)
directus  |     --------------------
directus  |     at Protocol._enqueue (/directus/node_modules/mysql/lib/protocol/Protocol.js:144:48)
directus  |     at Protocol.handshake (/directus/node_modules/mysql/lib/protocol/Protocol.js:51:23)
directus  |     at Connection.connect (/directus/node_modules/mysql/lib/Connection.js:116:18)
directus  |     at /directus/node_modules/knex/lib/dialects/mysql/index.js:56:18
directus  |     at new Promise (<anonymous>)
directus  |     at Client_MySQL.acquireRawConnection (/directus/node_modules/knex/lib/dialects/mysql/index.js:51:12)
directus  |     at create (/directus/node_modules/knex/lib/client.js:237:39)
directus  | npm notice 
directus  | npm notice New patch version of npm available! 7.24.0 -> 7.24.2
directus  | npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.24.2>
directus  | npm notice Run `npm install -g npm@7.24.2` to update!
directus  | npm notice
```

**Reply by gusdecool** [reply]
ah wait, i found how to make it works.
Directus have provided the docker compose example how to make it work https://github.com/directus/directus/blob/11d259b5851795e916b02339bb1dbd7c713e6d5a/docker-compose.yml#L71-L78

**Comment by eliasnemr** [comment]


**Reply by joselcvarela** [reply]
Hello @eliasnemr 
Since you are using `npm` I will assume you are running Directus 9.
Do you mind to open a new discussion? This thread is about Directus 8.
